client-src/elements/chromedash-feature-row.js

```diff
@@ -184,76 +142,53 @@ class ChromedashFeatureRow extends LitElement {
     return nothing;
   }
 
-  getEarliestReviewDate(feature) {
-    const featureConfigs = this.configs[feature.id] || [];
-    const allDates = featureConfigs.map(c => c.next_action).filter(d => d);
-    if (allDates.length > 0) {
-      allDates.sort();
-      return allDates[0];
-    }
-    return null;
-  }
-
-  getActiveOwners(feature) {
-    const featureConfigs = this.configs[feature.id] || [];
-    const allOwners = featureConfigs.map(c => c.owners).flat();
-    // TODO(jrobbins): Limit to only owners of active intents
-    let activeOwners = allOwners;
-    activeOwners = [...new Set(activeOwners)]; // de-dup.
-    activeOwners.sort();
-    return activeOwners;
+  isActiveGate(gate) {
+    return gate.state == 2 || gate.state == 3 || gate.state == 4;
   }
 
-  getActiveApprovals(feature) {
-    const featureApprovals = this.approvals[feature.id];
-    // TODO(jrobbins): Limit to only owners of active intents
-    const activeApprovals = featureApprovals;
-    return activeApprovals;
+  getActiveStages(feature) {
+    const featureGates = this.gates[feature.id] || [];
+    const activeGates = featureGates.filter(g => this.isActiveGate(g));
+    const activeStageIds = new Set(activeGates.map(g => g.stage_id));
+    const activeStagesAndTheirGates = [];
+    for (const activeStageId of activeStageIds) {
+      activeStagesAndTheirGates.push({
+        stageId: activeStageId,
+        gates: featureGates.filter(g => g.stage_id == activeStageId),
+      });
+    }
+    console.log(activeStagesAndTheirGates);
```

Comment:
Nit: more descriptive logging, e.g. console.log(activeStagesAndTheirGates is ...)

Reply:
Removed.  I had left that in by mistake.